src/django/pfb_analysis/management/commands/add_city_fips.py

```diff
@@ -0,0 +1,148 @@
+from django.core.management.base import BaseCommand
+
+import csv
+import logging
+import os
+import sys
+
+import boto3
+from botocore.exceptions import ClientError
+import fiona
+from fiona.crs import from_epsg
+
+logger = logging.getLogger(__name__)
+
+FIPS_CSV = 'national_places.csv'
+# local path to cached FIPS CSV
+FIPS_PATH = '/data/{csv}'.format(csv=FIPS_CSV)
+# S3 bucket where FIPS CSV should exist
+FIPS_BUCKET = 'pfb-public-documents'
+
+
+def add_city_fips(input_path, output_path, fips_csv):
+    logger.info("""Writing city FIPS Shapefile for {input}
+                to {output} using FIPS from {fips}""".format(input=input_path,
+                                                             output=output_path,
+                                                             fips=fips_csv))
+    # ensure Unicode will be handled properly
+    reload(sys)
+    sys.setdefaultencoding('utf8')
+
+    # map states to place names and their FIPS, for fast lookup
+    lookup = {}
+    with open(FIPS_PATH) as inf:
+        rdr = csv.DictReader(inf)
+        for place in rdr:
+            fips = '{state}{place}'.format(state=place['STATEFP'], place=place['PLACEFP'])
+            if not (place['STATE']) in lookup:
+                lookup[place['STATE']] = {place['PLACENAME'].upper(): fips}
+            else:
+                lookup[place['STATE']][place['PLACENAME'].upper()] = fips
+
+    with fiona.open(input_path) as shp:
+        schema = shp.schema.copy()
+        # CRS not set on input, so explicitly set it here
+        crs = from_epsg(4326)
+        # Add field to Shapefile properties
+        schema['properties']['city_fips'] = 'str:7'
+        with fiona.open(output_path, 'w', driver='ESRI Shapefile',
+                        schema=schema, crs=crs) as outshp:
+            place_count = len(shp)
+            found = 0
+            for feature in shp:
+                prop = feature.get('properties')
+                city = prop['city'].upper()
+                state_places = lookup.get(prop['state']).keys()
+                found_fips = None
+                if state_places:
+                    for place in state_places:
+                        if place.find(city) == 0:
+                            fips = lookup[prop['state']][place]
+                            logger.info("""Found place {place} for {city} in {state}
+                                with FIPS {fips}""".format(place=place, city=city,
+                                                           state=prop['state'], fips=fips))
```

Comment:
Thanks!  Showing both the `place` and the `city` should help with catching fuzzy matching errors, too.
NBD, but if this were
```
logger.info("Found place {place} for {city} in {state} "
            "with FIPS {fips}".format(
                place=place, city=city, state=prop['state'], fips=fips))
```
the line break wouldn't come through in the log message.